### PR TITLE
feat(profile): activate local account on profile

### PR DESCRIPTION
**Pull request by bodnara**
* local acount activation is performed in user-profile
* before this action had to be done in password-reset app

### CODE REVIEW

File: apps/user-profile/src/app/pages/settings-page/settings-local-account/settings-local-account.component.ts [hunk 1]
```diff
@@ -2,29 +2,49 @@ import { Component } from '@angular/core';
...
+    const config = getDefaultDialogConfig();
+    config.width = '550px';
+    config.data = {
+      userId: this.userId,
+      namespace: this.namespace,
+    };
+
+    const dialogRef = this.dialog.open(ActivateLocalAccountDialogComponent, config);
+    dialogRef.afterClosed().subscribe(() => {
+      void this.router.navigate(['/profile', '/settings', 'localAccount'], {
```
Comment (HejdaJakub) [inline apps/user-profile/src/app/pages/settings-page/settings-local-account/settings-local-account.component.ts:45]: `Error: Cannot match any routes. URL Segment: 'profile/%2Fsettings/localAccount'`
```suggestion
      void this.router.navigate(['/profile', 'settings', 'localAccount'], {
```

File: apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    [formGroup]="pwdForm"
+    [namespace]="data.namespace"
+    [language]="lang">
+  </perun-web-apps-password-form>
+</div>
+<div mat-dialog-actions>
+  <button mat-flat-button class="ml-auto mr-2" (click)="cancel()">
+    {{'DIALOGS.ACTIVATE_LOCAL_ACCOUNT.CANCEL'|customTranslate | translate}}
+  </button>
+  <button mat-flat-button color="accent" [disabled]="loading" (click)="activate()">
```
Comment (HejdaJakub) [inline apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.html:14]: Disable activate button when passwords don't match.
```suggestion
  <button mat-flat-button color="accent" [disabled]="loading  || pwdForm.invalid" (click)="activate()">
```

File: apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  cancel(): void {
+    this.dialogRef.close();
+  }
+
+  activate(): void {
+    this.loading = true;
+    const pwd: string = this.pwdForm.get('passwordCtrl').value as string;
+    this.userManager
+      .reservePasswordForUser(this.data.userId, this.data.namespace, pwd)
```
Comment (HejdaJakub) [inline apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.ts:56]: This method doesn't work correctly (at least for our use case probably?). Even if I type correct passwords, it always fails.
image

**Review by HejdaJakub — CHANGES_REQUESTED** [review]

File: apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.ts (hunk 3 above)
Comment (bodnara) [reply apps/user-profile/src/app/components/dialogs/activate-local-account-dialog/activate-local-account-dialog.component.ts:56]: NOTE: this functionality is instance specific

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**